test/SILGen/moveonly.swift

```diff
@@ -1000,3 +1004,2127 @@ struct AddressOnlySetterTester : ~Copyable {
+// MARK: Subscript //
+/////////////////////
+
+// MARK: Getter Only
+
+public struct LoadableSubscriptGetOnlyTester : ~Copyable {
+    subscript(_ i: Int) -> AddressOnlyProtocol {
+        get {
+            fatalError()
+        }
+    }
+}
+
+// CHECK-LABEL: sil [ossa] @$s8moveonly047testSubscriptGetOnly_BaseLoadable_ResultAddressE4_VaryyF : $@convention(thin) () -> () {
+// CHECK: [[BOX:%.*]] = alloc_box $
+// CHECK: [[PROJECT:%.*]] = project_box [[BOX]]
+//
+// The get call
+// CHECK: [[ACCESS:%.*]] = begin_access [read] [unknown] [[PROJECT]]
+// CHECK: [[MARK:%.*]] = mark_must_check [no_consume_or_assign] [[ACCESS]]
+// CHECK: [[LOAD_BORROW:%.*]] = load_borrow [[MARK]]
+// CHECK: [[TEMP:%.*]] = alloc_stack $AddressOnlyProtocol
+// CHECK: [[TEMP_MARK:%.*]] = mark_must_check [consumable_and_assignable] [[TEMP]]
+// CHECK: apply {{%.*}}([[TEMP_MARK]], {{%.*}}, [[LOAD_BORROW]])
+// CHECK: end_borrow [[LOAD_BORROW]]
+// CHECK: end_access [[ACCESS]]
+// CHECK: apply {{%.*}}([[TEMP_MARK]])
+// CHECK: destroy_addr [[TEMP_MARK]]
+//
+// Test the assignment
+// CHECK: [[M2_BOX:%.*]] = alloc_box ${
+// CHECK: [[M2_BORROW:%.*]] = begin_borrow [lexical] [[M2_BOX]]
+// CHECK: [[M2_PROJECT:%.*]] = project_box [[M2_BORROW]]
+// CHECK: [[ACCESS:%.*]] = begin_access [read] [unknown] [[PROJECT]]
+// CHECK: [[MARK:%.*]] = mark_must_check [no_consume_or_assign] [[ACCESS]]
+// CHECK: [[LOAD:%.*]] = load_borrow [[MARK]]
+// CHECK: apply {{%.*}}([[M2_PROJECT]], {{%.*}}, [[LOAD]])
+// CHECK: end_borrow [[LOAD]]
+// CHECK: end_access [[ACCESS]]
+// CHECK: [[M2_MARK:%.*]] = mark_must_check [no_consume_or_assign] [[M2_PROJECT]]
+// CHECK: end_borrow [[M2_BORROW]]
+// CHECK: destroy_value [[M2_BOX]]
+// CHECK: } // end sil function '$s8moveonly047testSubscriptGetOnly_BaseLoadable_ResultAddressE4_VaryyF'
+public func testSubscriptGetOnly_BaseLoadable_ResultAddressOnly_Var() {
+    var m = LoadableSubscriptGetOnlyTester()
+    m = LoadableSubscriptGetOnlyTester()
+    m[0].nonMutatingFunc()
+    let m2 = m[0]
+    _ = m2
+}
+
+// CHECK-LABEL: sil [ossa] @$s8moveonly047testSubscriptGetOnly_BaseLoadable_ResultAddressE4_LetyyF : $@convention(thin) () -> () {
+// CHECK: [[BOX:%.*]] = alloc_box $
+// CHECK: [[PROJECT:%.*]] = project_box [[BOX]]
+//
+// The get call
+// CHECK: [[MARK:%.*]] = mark_must_check [no_consume_or_assign] [[PROJECT]]
+// CHECK: [[LOAD_BORROW:%.*]] = load_borrow [[MARK]]
+// CHECK: [[TEMP:%.*]] = alloc_stack $AddressOnlyProtocol
+// CHECK: [[TEMP_MARK:%.*]] = mark_must_check [consumable_and_assignable] [[TEMP]]
+// CHECK: apply {{%.*}}([[TEMP_MARK]], {{%.*}}, [[LOAD_BORROW]])
+// CHECK: apply {{%.*}}([[TEMP_MARK]])
+// CHECK: destroy_addr [[TEMP_MARK]]
+// CHECK: end_borrow [[LOAD_BORROW]]
+// CHECK: } // end sil function '$s8moveonly047testSubscriptGetOnly_BaseLoadable_ResultAddressE4_LetyyF'
+public func testSubscriptGetOnly_BaseLoadable_ResultAddressOnly_Let() {
+    let m = LoadableSubscriptGetOnlyTester()
+    m[0].nonMutatingFunc()
```

Comment:
It looks like in this case the `nonMutatingFunc()` call still occurs inside of the borrow of `m`.

Reply:
This is not a result of this change since it is using a getter. I am pretty sure this is a result of us borrowing bases of accessors in an aggressive way which was part of the original implementation of accessors. IIRC @atrick ran into this when exploring non-escaping types.